src/Aspire.Hosting/Orchestrator/ParameterProcessor.cs

```diff
@@ -36,18 +45,29 @@ public async Task InitializeParametersAsync(IEnumerable<ParameterResource> param
 
         // If interaction service is available, we can handle unresolved parameters.
         // This will allow the user to provide values for parameters that could not be initialized.
-        if (interactionService.IsAvailable)
+        if (interactionService.IsAvailable && _unresolvedParameters.Count > 0)
         {
-            // All parameters have been processed, we can now handle unresolved parameters if any.
-            if (_unresolvedParameters.Count > 0)
+            if (waitForResolution)
+            {
+                // Handle unresolved parameters synchronously and wait for completion.
+                try
+                {
+                    await HandleUnresolvedParametersAsync().ConfigureAwait(false);
+                    logger.LogDebug("All unresolved parameters have been handled successfully.");
+                }
+                catch (Exception ex)
+                {
+                    logger.LogError(ex, "Failed to handle unresolved parameters.");
+                }
+            }
+            else
             {
                 // Start the loop that will allow the user to specify values for unresolved parameters.
-                _ = Task.Run(async () =>
+                _parameterResolutionTask = Task.Run(async () =>
```

Comment:
We don't need a branch here. 

var task = Task.Run(...);

if (waitForResolution)
{
   await task.ConfigureAwait(false);
}